modules/mauth-authenticator/src/test/scala/com/mdsol/mauth/scaladsl/utils/RequestAuthenticatorSpec.scala

```diff
@@ -72,6 +72,47 @@ class RequestAuthenticatorSpec extends FlatSpec with RequestAuthenticatorBaseSpe
     }
   }
 
+  it should "authenticate a valid request with V2 headers" in clientContext { (client) =>
+    //noinspection ConvertibleToMethodValue
+    (mockEpochTimeProvider.inSeconds _: () => Long).expects().returns(CLIENT_X_MWS_TIME_HEADER_VALUE.toLong + 3)
+    val authenticator = new RequestAuthenticator(client, mockEpochTimeProvider)
+
+    whenReady(authenticator.authenticate(getSimpleRequestV2)) { validationResult =>
+      validationResult shouldBe true
+    }
+  }
+
+  it should "authenticate a valid request with V2 headers if disabled V1" in clientContext { (client) =>
+    //noinspection ConvertibleToMethodValue
+    (mockEpochTimeProvider.inSeconds _: () => Long).expects().returns(CLIENT_X_MWS_TIME_HEADER_VALUE.toLong + 3)
+    val authenticator = new RequestAuthenticator(client, mockEpochTimeProvider, true)
+
+    whenReady(authenticator.authenticate(getSimpleRequestV2)) { validationResult =>
+      validationResult shouldBe true
+    }
+  }
+
+  it should "authenticate a valid request with all headers" in clientContext { (client) =>
+    //noinspection ConvertibleToMethodValue
+    (mockEpochTimeProvider.inSeconds _: () => Long).expects().returns(CLIENT_X_MWS_TIME_HEADER_VALUE.toLong + 3)
+    val authenticator = new RequestAuthenticator(client, mockEpochTimeProvider)
+
+    whenReady(authenticator.authenticate(getRequestWithAllHeaders)) { validationResult =>
+      validationResult shouldBe true
+    }
+  }
+
+  it should "not authenticate a request V1 header only if disabled V1" in {
```

Comment:
maybe this description reads better as: should reject a request with V1 headers when V1 is disabled

Reply:
updated as suggested